// angular2-flex/flex/flex.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { NgModule } from '@angular/core';
import { BREAKPOINTS, LayoutUtility, AttributeDirectiveFactory } from '../core';
export var FLEX_OPTIONS = [
    'grow',
    'initial',
    'auto',
    'none',
    'noshrink',
    'nogrow'
];
function getValue(value) {
    var attributeValue = value;
    if (!LayoutUtility.findIn(attributeValue, FLEX_OPTIONS)) {
        if (isNaN(attributeValue)) {
            attributeValue = '';
        }
    }
    return attributeValue;
}
var directives = [];
var generator = new AttributeDirectiveFactory();
BREAKPOINTS.forEach(function (breakPoint) {
    var name = breakPoint ? "flex-" + breakPoint : 'flex';
    directives.push(generator.generateDirective(name, breakPoint, getValue));
});
export var FlexModule = (function () {
    function FlexModule() {
    }
    FlexModule.forRoot = function () {
        return {
            ngModule: FlexModule,
            providers: []
        };
    };
    FlexModule = __decorate([
        NgModule({
            declarations: directives.slice(),
            imports: [],
            exports: directives.slice(),
        }), 
        __metadata('design:paramtypes', [])
    ], FlexModule);
    return FlexModule;
}());
